### PR TITLE
Added WCF Data Services deprecation message banner to index and overview

### PR DESCRIPTION
**Pull request by asanyaga**
## Summary

Adding a notice with deprecation information to the WCF Data Services index and overview pages.

Fixes #Issue_Number (if available)

### CODE REVIEW

**Comment by dnfadmin** [comment]
[CLA assistant check](https://cla.dotnetfoundation.org/dotnet/docs?pullRequest=22469) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.dotnetfoundation.org/dotnet/docs?pullRequest=22469) before we can accept your contribution.<br/><br/>:x: asanyaga [sign now](https://cla.dotnetfoundation.org/dotnet/docs?pullRequest=22469)<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.dotnetfoundation.org/check/dotnet/docs?pullRequest=22469) it.</sub>

**Review by mikepizzo — APPROVED** [review]
Looks good.

All -- WCF Data Services was removed from the Microsoft Download Centre 1/24/21. It would be great to get this into the docs asap.

@stevestein  -- Let us know if you need anything else from us in order to merge.

File: docs/framework/data/wcf/index.md [hunk 1]
```diff
@@ -10,6 +10,15 @@ ms.assetid: 73d2bec3-7c92-4110-b905-11bb0462357a
 
 # WCF Data Services 4.5
 
+> [!IMPORTANT]
+> WCF Data Services has been deprecated and is no longer available for download on the Microsoft Download Centre.
+> WCF Data Services supported earlier versions of the Microsoft OData (V1-V3) protocol only and has not been under active development. OData V1-V3 has been superseded by OData V4, which is an industry standard published by OASIS and ratified by ISO. OData V4 is supported through the OData V4 compliant core libraries available at [Microsoft.OData.Core](https://www.nuget.org/packages/Microsoft.OData.Core/) with support documentation available at [OData.Net](https://odata.github.io/odata.net) and the OData V4 service libraries available at [Microsoft.AspNetCore.OData](https://www.nuget.org/packages/Microsoft.AspNetCore.OData).
+>
+> [Restier](https://github.com/OData/RESTier) is the successor to WCF Data Services. RESTier helps you bootstrap a standardized, queryable HTTP-based REST interface in literally minutes.
```
Comment (gewarren) [inline docs/framework/data/wcf/index.md:17]:
```suggestion
> [RESTier](https://github.com/OData/RESTier) is the successor to WCF Data Services. RESTier helps you bootstrap a standardized, queryable, HTTP-based REST interface in minutes.
```

File: docs/framework/data/wcf/index.md [hunk 2]
```diff
@@ -10,6 +10,15 @@ ms.assetid: 73d2bec3-7c92-4110-b905-11bb0462357a
 
 # WCF Data Services 4.5
 
+> [!IMPORTANT]
+> WCF Data Services has been deprecated and is no longer available for download on the Microsoft Download Centre.
+> WCF Data Services supported earlier versions of the Microsoft OData (V1-V3) protocol only and has not been under active development. OData V1-V3 has been superseded by OData V4, which is an industry standard published by OASIS and ratified by ISO. OData V4 is supported through the OData V4 compliant core libraries available at [Microsoft.OData.Core](https://www.nuget.org/packages/Microsoft.OData.Core/) with support documentation available at [OData.Net](https://odata.github.io/odata.net) and the OData V4 service libraries available at [Microsoft.AspNetCore.OData](https://www.nuget.org/packages/Microsoft.AspNetCore.OData).
```
Comment (gewarren) [inline docs/framework/data/wcf/index.md:15]:
```suggestion
> WCF Data Services supported earlier versions of the Microsoft OData (V1-V3) protocol only and has not been under active development. OData V1-V3 has been superseded by OData V4, which is an industry standard published by OASIS and ratified by ISO. OData V4 is supported through the OData V4 compliant core libraries available at [Microsoft.OData.Core](https://www.nuget.org/packages/Microsoft.OData.Core/). Support documentation is available at [OData.Net](https://odata.github.io/odata.net), and the OData V4 service libraries are available at [Microsoft.AspNetCore.OData](https://www.nuget.org/packages/Microsoft.AspNetCore.OData).
```

**Review by gewarren — COMMENTED** [review]
It would be best to put the boilerplate text into an include file, since it's reused here in a couple places and will potentially be added to other articles going forward. LMK if you need help doing that. Thanks for submitting this PR!

**Comment by asanyaga** [comment]
> It would be best to put the boilerplate text into an include file, since it's reused here in a couple places and will potentially be added to other articles going forward. LMK if you need help doing that. Thanks for submitting this PR!

@gewarren thank you very much for reviewing and for the feedback. I would appreciate some guidance on how to use an include file for this, thank you.

**Comment by gewarren** [comment]
@asanyaga Create a new markdown file in [this folder](https://github.com/dotnet/docs/tree/master/includes). Put the `[!IMPORTANT]` block in the file. Then to reference it from each of the above files, add a line like this:

`[!INCLUDE[wcf-deprecated](~/includes/wcf-deprecated.md)]`

**Comment by asanyaga** [comment]
> @asanyaga Create a new markdown file in [this folder](https://github.com/dotnet/docs/tree/master/includes). Put the `[!IMPORTANT]` block in the file. Then to reference it from each of the two data\wcf\ files, add a line like this:
> 
> `[!INCLUDE[wcf-deprecated](~/includes/wcf-deprecated.md)]`

@gewarren thanks for the tip. i've made the changes.

**Review by gewarren — APPROVED** [review]
Looks great. Thank you @asanyaga !

**Comment by asanyaga** [comment]
> Looks great. Thank you @asanyaga !

@gewarren thank you!

**Comment by asanyaga** [comment]
@gewarren, I have a question. We would like to add the deprecation notice to all the pages on the WCF documentation. Should I create a new branch for that or should I just push to this branch? thanks.

**Comment by tdykstra** [comment]
In that case it should also be added to the WCF tools in the [docs/core/additional-tools folder](https://github.com/dotnet/docs/tree/master/docs/core/additional-tools).

**Comment by gewarren** [comment]
@asanyaga Please create a new branch. Thank you!

**Comment by asanyaga** [comment]
> In that case it should also be added to the WCF tools in the [docs/core/additional-tools folder](https://github.com/dotnet/docs/tree/master/docs/core/additional-tools).

@tdykstra, the WCF Web Services tools will not be affected by the deprecation of the WCF Data Services so it will not be necessary to add the notice there, thanks.

**Pull request merged**